base_layer/wallet_ffi/src/error.rs

```diff
@@ -363,3 +363,29 @@ impl From<SchnorrSignatureError> for LibWalletError {
         }
     }
 }
+
+#[derive(Debug, Error, PartialEq)]
+pub enum TransactionError {
+    #[error("The transaction has an incorrect status: `{0}`")]
+    StatusError(String),
+    #[error("The transaction has the wrong number of kernels: `{0}`")]
+    KernelError(String),
+}
+
+/// This implementation maps the internal TransactionError to a set of LibWalletErrors.
+/// The mapping is explicitly managed here.
+impl From<TransactionError> for LibWalletError {
+    fn from(v: TransactionError) -> Self {
+        error!(target: LOG_TARGET, "{}", format!("{:?}", v));
```

Comment:
```suggestion
        error!(target: LOG_TARGET, "{}", v);
```
However, I don't think an `error!` log level makes sense for all errors. It feels like an unexpected side effect to log in a conversion function, so I'd say rather log at the call site.

Reply:
Thanks, it's done similarly for other errors. @philipr-za what do you suggest to do here?

Reply:
It's done the same way for all the other errors here in wallet_ffi, so will keep it consistent for now. If they all need to be refactored then I'll add that in a separate PR.

Reply:
So the format was left over from the old error derive macro we used so that can definitely go. 

I am not sure I agree about the logging here, these errors are all show stoppers for the FFI call that elicits them and indicate that the client library is doing something wrong, I think those should be logged at the highest level and are not worth trying to get too specific about.